src/s3/commands/uploadFile.ts

```diff
@@ -104,104 +140,244 @@ export async function uploadFileCommand(
-            'Uploading file {0} to {1}', 
-            fileName, 
-            destinationPath), 
-        outputChannel)
-
-        const request = {
-            bucketName: bucket.Name!,
-            key: key,
-            fileLocation: file,
-            fileSizeBytes: fileSizeBytes(file),
-            s3Client,
-            window: window,
+    let failedRequests = await uploadBatchOfFiles(uploadRequests, window, outputChannel)
+    //let failedRequests = await uploadWithProgress(uploadRequests, undefined, outputChannel)
+
+    const completedRequests = uploadRequests.length - failedRequests.length
+    showOutputMessage(
+        localize(
+            'AWS.s3.uploadFile.success',
+            'Successfully uploaded {0}/{1} files',
+            completedRequests,
+            uploadRequests.length
+        ),
+        outputChannel
+    )
+
+    while (failedRequests.length > 0) {
+        const failedKeys = failedRequests.map(request => request.key)
+        getLogger().error(`List of requests failed to upload:\n${failedRequests.toString().split(',').join('\n')}`)
+
+        if (failedRequests.length > 5) {
+            showOutputMessage(
+                localize(
+                    'AWS.s3.uploadFile.failed',
+                    'Failed uploads:\n{0}\nSee logs for full list of failed items',
+                    failedKeys.toString().split(',').slice(0, 5).join('\n')
+                ),
+                outputChannel
+            )
         }
+        if (failedRequests.length > 0) {
+            showOutputMessage(
+                localize(
+                    'AWS.s3.uploadFile.failed',
+                    'Failed uploads:\n{0}',
+                    failedKeys.toString().split(',').join('\n')
+                ),
 
-        await uploadWithProgress(request)
+                outputChannel
+            )
+        }
 
-        showOutputMessage(localize(
-            'AWS.s3.uploadFile.success', 
-            'Successfully uploaded file {0} to {1}', 
-            fileName, 
-            bucket.Name), 
-        outputChannel)
-        telemetry.recordS3UploadObject({ result: 'Succeeded' })
-        recordAwsRefreshExplorer()
-        commands.execute('aws.refreshAwsExplorer')
-        return
-    } catch (e) {
-        getLogger().error(`Failed to upload file from ${file} to ${destinationPath}: %O`, e)
-        showErrorWithLogs(localize('AWS.s3.uploadFile.error.general', 'Failed to upload file {0}', fileName), window)
-        telemetry.recordS3UploadObject({ result: 'Failed' })
-        return
+        //at least one request failed
+        const response = await window.showErrorMessage(
+            `${failedRequests.length}/${uploadRequests.length} failed.`,
+            'Try again',
+            'Continue'
+        )
+        if (response === 'Try again') {
+            failedRequests = await uploadBatchOfFiles(failedRequests, window, outputChannel)
+        } else {
+            break
+        }
```

Comment:
I'm assuming that the while loop is here so this message can keep executing? Move this to a `.then` and remove the while loop. That way, the code won't keep executing and won't block on this response.

Reply:
The while loop is so that the message keeps popping every time there is at least a failed request